src/main/java/org/traccar/protocol/TranSyncProtocolDecoder.java

```diff
@@ -0,0 +1,218 @@
+        simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        try {
+            return simpleDateFormat.parse(dataString);
+
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    protected void setDeviceAlert(Position position, int alert) {
+        switch (alert) {
+            case 10:
+                position.set(Position.ALARM_SOS, true);
+                break;
+            case 11:
+                position.set(Position.ALARM_SOS, false);
+                break;
+            case 16:
+                position.set(Position.KEY_EVENT, 16);
+                break;
+            case 3:
+                position.set("distressCutOff", true);
+                break;
+            case 22:
+                position.set("tilt", true);
+                break;
+            case 9:
+                position.set(Position.KEY_EVENT, 9);
+            case 17:
+                position.set(Position.ALARM_OVERSPEED, true);
+                break;
+            case 13:
+                position.set(Position.ALARM_BRAKING, true);
+                break;
+            case 14:
+                position.set(Position.ALARM_ACCELERATION, true);
+                break;
+            case 15:
+                position.set(Position.KEY_EVENT, 15);
+                break;
+            case 23:
+                position.set(Position.ALARM_ACCIDENT, true);
+                break;
+            case 12:
+                position.set(Position.KEY_EVENT, 12);
+                break;
+            case 6:
+                position.set(Position.ALARM_POWER_RESTORED, true);
+                break;
+            case 4:
+                position.set(Position.ALARM_LOW_BATTERY, true);
+                break;
+            case 5:
+                position.set(Position.KEY_EVENT, 5);
+                break;
+
+        }
+
+    }
+
+    private void setNetwork(Position position, int lacInt, int cellIdInt, int mobileNetworkCode, int gsmSignalStrength){
+        CellTower cellTower = CellTower.fromLacCid(getConfig(), lacInt, cellIdInt);
+        cellTower.setMobileNetworkCode(mobileNetworkCode);
+        cellTower.setSignalStrength(gsmSignalStrength);
+        position.setNetwork(new Network(cellTower));
+
+    }
+
+    private String getTrackerModel(int mask){
+
+        switch (mask) {
+            case 1:
+                return  "basic";
+            case 2:
+                return "asset";
+            case 3:
+                return "bike";
+
+            case 4:
+                return "serial";
+
+            case 5:
+                return "obd";
+            case 6:
+                return "l1";
+            case 7:
+                return "ais";
+            default:
+                return "unknown";
+        }
+
+    }
+
+    private Position parseData(Channel channel, SocketAddress remoteAddress, ByteBuf buf) {
+        if (ByteBufUtil.hexDump(buf, 0, 2).equalsIgnoreCase("3A3A")) {
```

Comment:
It doesn't make sense to convert into a string if you can just read an integer value.